File: src/test/java/UtilityTest.java
import de.floribe2000.warships_java.direct.account.PlayersPersonalDataFull;
import de.floribe2000.warships_java.direct.api.ApiBuilder;
import de.floribe2000.warships_java.direct.api.Region;
import de.floribe2000.warships_java.direct.encyclopedia.Warships;
import de.floribe2000.warships_java.direct.encyclopedia.WarshipsRequest;
import de.floribe2000.warships_java.utilities.EncyclopediaRequestService;
import de.floribe2000.warships_java.utilities.PlayerRequestService;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;

import java.io.FileInputStream;
import java.io.IOException;
import java.util.Properties;

public class UtilityTest {

    private final String apiKey;

    public UtilityTest() throws IOException {
        Properties PROPERTIES = new Properties();
        PROPERTIES.load(new FileInputStream("Warships.properties"));
        apiKey = PROPERTIES.getProperty("APIKEY");
    }

    @BeforeClass
    public static void prepareTests() throws IOException {
        ApiBuilder.shutdown();
    }

    @AfterClass
    public static void shutDownTests() throws IOException {
        ApiBuilder.shutdown();
    }

    @Test
    public void testShipRequestService() {
        EncyclopediaRequestService.initialize(apiKey);
        Warships warships = WarshipsRequest.createRequest().region(Region.EU).fetch();
        assert warships.getStatus().get() : "Invalid response status";
        Warships fullList = EncyclopediaRequestService.requestFullWarshipsList(Region.EU);
        assert fullList.getStatus().get() : "Invalid response status from combined list";
        assert fullList.getData().size() == warships.getMeta().getTotal() :
                "Size of retrieved list does not match expected size. Expected " + warships.getMeta().getTotal() + ", got " + fullList.getData().size();
    }

    @Test
    public void testPlayerRequestService() {
        PlayerRequestService.initialize(apiKey);
        String playerName = "floribe2000";
        PlayersPersonalDataFull player = PlayerRequestService.requestPlayersPersonalData(playerName, Region.EU);
        assert player.getStatus().get() : "Invalid response status";
        assert player.getData().size() > 0 : "Empty response";
        String playerId = player.getData().keySet().iterator().next();
        assert player.getData().get(playerId).getNickname().equals(playerName) :
                "Result does not match search name. Expected " + playerName + ", got " + player.getData().get(playerId).getNickname();
    }
}

File: src/test/kotlin/UtilityTest.kt
import de.floribe2000.warships_java.direct.api.typeDefinitions.Language
import de.floribe2000.warships_java.direct.api.typeDefinitions.Region
import de.floribe2000.warships_java.direct.encyclopedia.WarshipsRequest.Companion.createRequest
import de.floribe2000.warships_java.utilities.EncyclopediaRequestService
import de.floribe2000.warships_java.utilities.EncyclopediaRequestService.requestFullWarshipsList
import de.floribe2000.warships_java.utilities.PlayerRequestService
import de.floribe2000.warships_java.utilities.PlayerRequestService.requestPlayersPersonalData
import org.junit.jupiter.api.Test
import java.io.FileInputStream
import java.util.*

class UtilityTest {
    private val apiKey: String

    @Test
    fun testShipRequestService() {
        EncyclopediaRequestService.initialize(apiKey)
        val warships = createRequest().region(Region.EU).fetch()
        assert(warships.status.get()) { "Invalid response status" }
        val fullList = requestFullWarshipsList(Region.EU)
        assert(fullList.status.get()) { "Invalid response status from combined list" }
        assert(fullList.data!!.size == warships.meta.total) { "Size of retrieved list does not match expected size. Expected " + warships.meta.total + ", got " + fullList.data!!.size }
    }

    @Test
    fun testGermanWarshipsList() {
        EncyclopediaRequestService.initialize(apiKey)
        val warships = createRequest().region(Region.EU).fetch()
        assert(warships.status.get()) { "Invalid response status" }
        val germanList = requestFullWarshipsList(Region.EU, Language.ENGLISH)
        assert(germanList.status.get()) { germanList }
        assert(germanList.data!!.size == warships.meta.total) {
            """
     Size of retrieved list does not match expected size. Expected ${warships.meta.total}, got ${germanList.data!!.size}
     Json Response:
     $germanList
     """.trimIndent()
        }
    }

    @Test
    fun testPlayerRequestService() {
        PlayerRequestService.initialize(apiKey)
        val playerName = "floribe2000"
        val player = requestPlayersPersonalData(playerName, Region.EU)
        assert(player.status.get()) { "Invalid response status" }
        assert(player.data.isNotEmpty()) { "Empty response" }
        val playerId = player.data.keys.first()
        assert(
            (player.data[playerId]
                ?: error("")).nickname == playerName
        ) { "Result does not match search name. Expected " + playerName + ", got " + player.data[playerId]!!.nickname }
    }

    init {
        val properties = Properties()
        properties.load(FileInputStream("Warships.properties"))
        apiKey = properties.getProperty("APIKEY")
    }
}